sklearn/metrics/ranking.py

```diff
@@ -83,8 +94,13 @@ def auc(x, y, reorder=False):
         raise ValueError('At least 2 points are needed to compute'
                          ' area under curve, but x.shape = %s' % x.shape)
 
+    if reorder != 'deprecated':
+        warnings.warn("The `reorder` parameter has been deprecated "
+                      "in version 0.20 and will be removed in 0.22.",
```

Comment:
Best practices: the warning ideally:
* explains why there is a warning (maybe here the doc + release notes is enough in this case)
* what to do to get rid of it. In this case say it is recommended to not set reorder (same behaviour as reorder=False) and make sure to have x and y ordered (maybe add as the points of a curve. I am not sure how to phrase this better ...)